Project Web Mercator Tiles to Sphere with ShaderMaterial

cmurphy23 opened the pull request:
This PR is for #74, where I also wrote some comments about my work as I went.

The changes are primarily to MapSphereNode, specifically swapping the material for a ShaderMaterial that calculates the color of a given position by looking up the corresponding pixel on the web mercator tile image. In addition, we change the size of the tiles so that their position properly matches the coordinates they are supposed to show. This method, while not identical, was inspired by a very similar approach used by CesiumJS.

Added the `applyTexture` function for MapNode base class so that MapSphereNode could override it load the texture into the ShaderMaterial.

Also added some utility functions to UnitsUnitls.

I wasn't sure whether to update the build/ folder or not, I can remove those changes from the PR if necessary.

Before:
![image](https://github.com/tentone/geo-three/assets/3690500/d4f99d2e-03e1-4abd-9a13-b66cae735101)

After: 
![image](https://github.com/tentone/geo-three/assets/3690500/13d46fbf-6a3d-49bd-88a0-b80a6b10c11e)

tentone commented on source/utils/UnitsUtils.ts:
```diff
@@ -140,4 +145,31 @@ export class UnitsUtils
 		return (color.r * 255.0 * 65536.0 + color.g * 255.0 * 256.0 + color.b * 255.0) * 0.1 - 10000.0;
 	}
+
+	public static getTileSize(zoom: number): number
+	{
+		const maxExtent = UnitsUtils.MERCATOR_MAX_EXTENT;
+		const numTiles = Math.pow(2, zoom);
+		return 2 * maxExtent / numTiles;	
+	}
+
+	public static tileBounds(zoom: number, x: number, y: number): number[]
```
Please add docs for these parameters.

tentone reviewed (approved):
Hello

Check documentation for auxiliary methods please, other than that code seems alright.

cmurphy23 commented:
Okay, I added some documentation for those methods, I reverted the build folder to before my changes, and I fixed some weird indentation on the shader code.

FengFengmomo commented:
i have modify code as yours but there still some misalignment 
![image](https://github.com/tentone/geo-three/assets/12838106/d4879dde-09ac-4192-93bf-aea8db7e2a04)

cmurphy23 commented:
I'm not sure if this PR is the right place to discuss, but the problem you're having is because the tile bounds expected by the shader code don't match the tile bounds used for the tile's geometry, specifically in the Y/latitutde direction. I recommend using identical calculations for both.
